packages/vx-axis/src/axis/Axis.tsx

```diff
@@ -46,20 +46,21 @@ export default function Axis<Scale extends AxisScale>({
     horizontal,
   );
 
-  const ticks = (tickValues ?? getTicks(scale, numTicks))
+  const filteredTickValues = (tickValues ?? getTicks(scale, numTicks))
     .map((value, index) => ({ value, index }))
-    .filter(({ value }) => !hideZero || (value !== 0 && value !== '0'))
-    .map(({ value, index }) => {
-      const scaledValue = coerceNumber(tickPosition(value));
+    .filter(({ value }) => !hideZero || (value !== 0 && value !== '0'));
 
-      return {
-        value,
-        index,
-        from: createPoint({ x: scaledValue, y: 0 }, horizontal),
-        to: createPoint({ x: scaledValue, y: tickLength * tickSign }, horizontal),
-        formattedValue: format(value, index),
-      };
-    });
+  const ticks = filteredTickValues.map(({ value, index }) => {
+    const scaledValue = coerceNumber(tickPosition(value));
+
+    return {
+      value,
+      index,
+      from: createPoint({ x: scaledValue, y: 0 }, horizontal),
+      to: createPoint({ x: scaledValue, y: tickLength * tickSign }, horizontal),
+      formattedValue: format(value, index, filteredTickValues),
```

Comment:
have been wanting to add this for a while (will update the PR description): we now pass all ticks as a third argument (common d3 pattern) so that users can determine if a tick is first/last, etc. with `index` only you couldn't ever determine this because you may not know the exact # ticks unless you specify `tickValues`